src/bin/profiling.rs

```diff
@@ -27,11 +23,11 @@ mod imp {
     const PROFILE_ACTIVE: &'static [u8] = b"prof.active\0";
     const PROFILE_DUMP: &'static [u8] = b"prof.dump\0";
 
-    struct DumpFileNameGuard(Option<Vec<u8>>);
+    struct DumpPathGuard(Option<Vec<u8>>);
 
-    impl DumpFileNameGuard {
-        fn from_cstring(s: Option<CString>) -> DumpFileNameGuard {
-            DumpFileNameGuard(s.map(|s| s.into_bytes_with_nul()))
+    impl DumpPathGuard {
+        fn from_cstring(s: Option<CString>) -> DumpPathGuard {
+            DumpPathGuard(s.map(|s| s.into_bytes_with_nul()))
```

Comment:
What is the name of the dump file if `s` is `Some("")`, working directory + auto-generated name ?